dusty/compiler/port_spec/__init__.py

```diff
@@ -52,12 +61,15 @@ def get_port_spec_document(expanded_active_specs, docker_vm_ip):
         port_spec['docker_compose'][app_name] = []
         container_ports = set()
         for host_forwarding_spec in app_spec['host_forwarding']:
-            add_full_addresses(host_forwarding_spec, host_full_addresses)
-            add_container_ports(host_forwarding_spec, container_ports)
+            # These functions are just used for validating the set of specs all works together
+            _add_full_addresses(host_forwarding_spec, host_full_addresses)
+            _add_container_ports(host_forwarding_spec, container_ports)
+            if host_forwarding_spec['type'] == 'stream':
+                _add_stream_host_port(host_forwarding_spec, stream_host_ports)
```

Comment:
`stream_host_ports.add(host_forwarding_spec['host_port'[)` should be fine after we validate for the conflict right?